Path related failures on Windows when running tests

When running `npm test` on Windows, I get the following errors:

```
  1) CLIEngine executeOnFiles() should return a warning when an explicitly given file is ignored:
     AssertionError: expected 'C:\\tmp\\eslint\\fixtures\\passing.js' to equal '\\tmp\\eslint\\fixtures\\passing.js'
      at Context.<anonymous> (C:\Users\BYK\Documents\Projects\eslint\eslint\tests\lib\cli-engine.js:532:20)

  2) Config findLocalConfigFiles() should return the path when an .eslintrc file is found:
     AssertionError: expected 'C:\\tmp\\eslint\\fixtures\\config-hierarchy\\broken\\.eslintrc' to equal '\\tmp\\eslint\\fixtures\\config-hierarchy\\broken\\.eslintrc'
      at Context.<anonymous> (C:\Users\BYK\Documents\Projects\eslint\eslint\tests\lib\config.js:115:20)

  3) Config findLocalConfigFiles() should return the path when a package.json file is found:
     AssertionError: expected 'C:\\tmp\\eslint\\fixtures\\config-hierarchy\\broken\\package.json' to equal '\\tmp\\eslint\\fixtures\\config-hierarchy\\broken\\package.json'
```

This is happening on latest master with Node 4.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27200159-path-related-failures-on-windows-when-running-tests?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/tests/lib/cli-engine.js
+++ b/tests/lib/cli-engine.js
@@ -18,12 +18,13 @@ var assert = require("chai").assert,
     leche = require("leche"),
     rules = require("../../lib/rules"),
     Config = require("../../lib/config"),
-    fs = require("fs");
+    fs = require("fs"),
+    os = require("os");
 
 require("shelljs/global");
 proxyquire = proxyquire.noCallThru().noPreserveCache();
 
-/* global tempdir, mkdir, rm, cp */
+/* global mkdir, rm, cp */
 
 //------------------------------------------------------------------------------
 // Tests
@@ -56,7 +57,7 @@ describe("CLIEngine", function() {
 
     // copy into clean area so as not to get "infected" by this project's .eslintrc files
     before(function() {
-        fixtureDir = tempdir() + "/eslint/fixtures";
+        fixtureDir = os.tmpdir() + "/eslint/fixtures";
         mkdir("-p", fixtureDir);
         cp("-r", "./tests/fixtures/.", fixtureDir);
     });

--- a/tests/lib/cli.js
+++ b/tests/lib/cli.js
@@ -16,6 +16,7 @@ var assert = require("chai").assert,
     sinon = require("sinon"),
     leche = require("leche"),
     fs = require("fs"),
+    os = require("os"),
     sh = require("shelljs"),
     proxyquire = require("proxyquire");
 
@@ -42,7 +43,7 @@ describe("cli", function() {
 
     // copy into clean area so as not to get "infected" by this project's .eslintrc files
     before(function() {
-        fixtureDir = sh.tempdir() + "/eslint/fixtures";
+        fixtureDir = os.tmpdir() + "/eslint/fixtures";
         sh.mkdir("-p", fixtureDir);
         sh.cp("-r", "./tests/fixtures/.", fixtureDir);
     });

--- a/tests/lib/config.js
+++ b/tests/lib/config.js
@@ -12,6 +12,7 @@
 var assert = require("chai").assert,
     path = require("path"),
     fs = require("fs"),
+    os = require("os"),
     Config = require("../../lib/config"),
     sinon = require("sinon"),
     proxyquire = require("proxyquire");
@@ -19,7 +20,7 @@ var assert = require("chai").assert,
 require("shelljs/global");
 proxyquire = proxyquire.noCallThru().noPreserveCache();
 
-/* global tempdir, mkdir, rm, cp */
+/* global mkdir, rm, cp */
 
 
 /**
@@ -75,7 +76,7 @@ describe("Config", function() {
 
     // copy into clean area so as not to get "infected" by this project's .eslintrc files
     before(function() {
-        fixtureDir = tempdir() + "/eslint/fixtures";
+        fixtureDir = os.tmpdir() + "/eslint/fixtures";
         mkdir("-p", fixtureDir);
         cp("-r", "./tests/fixtures/config-hierarchy", fixtureDir);
     });

